Tests/ML/models/test_scalar_model.py

```diff
@@ -337,7 +346,36 @@ def test_runner1(test_output_dirs: OutputFolderForTests) -> None:
     assert config.non_image_feature_channels == ["label"]
     assert str(config.outputs_folder).startswith(output_root)
     assert (config.logs_folder / LOG_FILE_NAME).exists()
+    # Check that we saved one checkpoint every second epoch and that we kept only that last 2 and that last.ckpt has
+    # been renamed to best.ckpt
+    assert len(os.listdir(config.checkpoint_folder)) == 3
+    assert (config.checkpoint_folder / str(RECOVERY_CHECKPOINT_FILE_NAME + "_epoch=3" + CHECKPOINT_SUFFIX)).exists()
+    assert (config.checkpoint_folder / str(RECOVERY_CHECKPOINT_FILE_NAME + "_epoch=5" + CHECKPOINT_SUFFIX)).exists()
+    assert (config.checkpoint_folder / BEST_CHECKPOINT_FILE_NAME_WITH_SUFFIX).exists()
+
 
+@pytest.mark.skipif(common_util.is_windows(), reason="Has OOM issues on windows build")
+def test_runner_restart(test_output_dirs: OutputFolderForTests) -> None:
+    """
+    Test if starting training from a folder where the checkpoints folder already has recovery checkpoints picks up
+    that it is a recovery run. Also checks that we update the start epoch in the config at loading time.
```

Comment:
I don't see where these statements are reflected in asserts? to me, it seems this test does not provide more than test_runner1

Reply:
Because here we test that we have only two checkpoints 3 and 5. If the run was restarting from epoch 0 we would have also a checkpoint for epoch=1. But i can probably make use for the `get_total_number_of_training_epochs` as you pointed out below to make that more explicit.